### PR TITLE
Slack notification

### LINKED ISSUE #118: Add Slack integration for build alerts
## Type of Issue:

- [ ] Bug: Something isn't working
- [x] Enhancement: New feature or request
- [ ] Question: Further information is requested (I have a question)

#### Issue Creator Checklist

- [x] The appropriate label (bug, enhancement, question) has been attached to this issue.
- [x] This issue has been thoroughly documented below; a developer should be able to understand the issue by reading it.

## Background

Users of downstream forks of this repo have expressed interest in getting Slack notifications when important GitHub Actions workflows fail.  Since Slack is widely used at CMS and in general, and since the notification logic will probably be adaptable to other communication platforms, this is a good candidate to add to the quickstart.  This issue/enhancement is to add that support.

Users care about when the Deploy workflow fails.  Projects/forks shouldn't be required to configure slack notifications; if I'm a user of this quickstart, I shouldn't need to configure Slack if I can't or don't want to.

The AC below should lay out the specific ask in detail, but in general... this issue is asking for optional slack integration on a per-branch basis.  That last part is important:  if we get notified on *all* branches, including dev branches, alert fatigue is inevitable.

## AC:

- A pattern for configuring slack notifications for the Deploy workflow is implemented.
- Slack notifications are optional... a user/fork of this repo does not need to configure Slack notifications if they can't or don't wish to.
- A pattern for configuring slack notifications on a per-branch/stage basis is implemented... a user/fork of this repo should be able to turn on Slack notifications, but only for branches that are deemed important.
- The configured slack channel is notified when the Deploy workflow fails.

### PR DESCRIPTION
**Pull request by joeaero19**
## Purpose
This PR is to integrate slack notification capability into master.
Currently, notifications are not been sent out upon job failure. We need to be alerted when this occur. We also need to give this capability to all other projects who pull from the quickstart

#### Linked Issues to Close

Closes #118 

## Approach
 
This change will resolve this the issue by sending alert on the associated slack channel.

## Learning

https://github.com/ravsamhq/notify-slack-action

## Assorted Notes/Considerations

N/A

#### Pull Request Creator Checklist

- [x] This PR has an associated issue or issues.
- [x] The associated issue(s) are linked above.
- [x] This PR meets all acceptance criteria for those issues.
- [x] This PR and linked issue(s) are adequately documented
- [x] This PR and linked issues(s) are a complete description of the changeset; an individual or team should be able to understand the issue(s) and changes by reading through this PR and it's links, with no further interaction.
- [x] Someone has been assigned this PR.
- [x] At least one person has been marked as reviewer on this PR.

#### Pull Request Reviewer/Assignee Checklist

- [x] This PR has an associated issue or issues.
- [x] The associated issue(s) are linked above.
- [x] This PR meets all acceptance criteria for those issues.
- [x] This PR and linked issue(s) are adequately documented
- [x] This PR and linked issues(s) are a complete description of the changeset; an individual or team should be able to understand the issue(s) and changes by reading through this PR and it's links, with no further interaction.

### CODE REVIEW

**Comment by mdial89f** [comment]
Definitley SQUASH and merge

File: .github/build_vars.sh [hunk 1]
```diff
@@ -5,6 +5,8 @@ var_list=(
   'AWS_SECRET_ACCESS_KEY'
   'AWS_DEFAULT_REGION'
   'STAGE_PREFIX'
+  'SLACK_INTEGRATION'
```
Comment (mdial89f) [inline .github/build_vars.sh:8]: This can be removed as it's not used

File: README.md [hunk 2]
```diff
@@ -6,18 +6,6 @@ A serverless form submission application built and deployed to AWS with the Serv
 
 ![Architecture Diagram](./.images/architecture.png?raw=true)
 
-## Local Dev
```
Comment (mdial89f) [inline README.md:9]: I think you've inadvertently removed a bunch of documentation.

**Review by mdial89f — CHANGES_REQUESTED** [review]
Hey I left two comments on the PR.
One is for a variable that can be removed.
The other (more important) is documentation that's been removed accidentally.

My recommendation is to make a new branch off master, and copy over the exact changes for the PR... that way you don't have to fool around with getting the README docs back in.

But in any case, that's what I see

File: .github/build_vars.sh (hunk 1 above)
Comment (joeaero19) [reply .github/build_vars.sh:8]: done

File: README.md (hunk 2 above)
Comment (joeaero19) [reply README.md:9]: hmmm. from the readme?

Comment (joeaero19) [reply README.md:9]: let me fix that

Comment (joeaero19) [reply README.md:9]: done

Comment (mdial89f) [reply README.md]: @joeaero19 When I look at the diff of the README in the Files Changed section, there's still quite a bit of stuff being removed.
I recommend you make a new branch (or reset this one) off of master, and add the handful of changes you need to.

Comment (joeaero19) [reply README.md]: ok

Comment (joeaero19) [reply README.md]: I compared with master readme file and added the diff.

**Review by mdial89f — APPROVED** [review]

**Pull request merged**